src/builder.test.ts

```diff
@@ -0,0 +1,398 @@
+import { describe, it, expect } from "vitest";
```

Comment:
This file is just a conglomerate of all the old test files that got deleted, just tweaked to use the new builder pattern.